src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,185 @@ into grouped
+                        HidePasswords = false,
+                        Manage = false
+                    })
+                    .ToList();
+                dbContext.CollectionUsers.AddRange(newCollectionUsers);
+
+                // Update OrganizationUser to clear AccessAll flag
+                organizationUser.AccessAll = false;
+            }
+
+            // Save changes for Step 2
+            await dbContext.SaveChangesAsync();
+
+            // Step 3: For all OrganizationUsers with Manager role or 'EditAssignedCollections' permission
+            // update their existing CollectionUser rows and insert new rows with [Manage] = 1
+            // and finally update all OrganizationUsers with Manager role to User role
+            var managerOrgUsers = dbContext.OrganizationUsers
+                .Where(ou =>
+                    ou.OrganizationId == organizationId &&
+                    (ou.Type == OrganizationUserType.Manager ||
+                     (ou.Permissions != null && EF.Functions.Like(ou.Permissions, "%\"editAssignedCollections\":true%"))))
+                .ToList();
+            var managerOrgUsersIds = managerOrgUsers
+                .Select(ou => ou.Id)
+                .ToList();
+
+            // Update CollectionUser rows with Manage = true
+            dbContext.CollectionUsers
+                .Where(cu => managerOrgUsersIds.Contains(cu.OrganizationUserId))
+                .ToList()
+                .ForEach(cu =>
+                {
+                    cu.ReadOnly = false;
+                    cu.HidePasswords = false;
+                    cu.Manage = true;
+                });
+
+            // Insert rows into CollectionUser with Manage = true
+            foreach (var manager in managerOrgUsers)
+            {
+                var newCollectionUsersWithManage = (from cg in dbContext.CollectionGroups
+                                                    join gu in dbContext.GroupUsers on cg.GroupId equals gu.GroupId
+                                                    where gu.OrganizationUserId == manager.Id &&
+                                                          !dbContext.CollectionUsers.Any(cu =>
+                                                              cu.CollectionId == cg.CollectionId &&
+                                                              cu.OrganizationUserId == manager.Id)
+                                                    select new CollectionUser
+                                                    {
+                                                        CollectionId = cg.CollectionId,
+                                                        OrganizationUserId = manager.Id,
+                                                        ReadOnly = false,
+                                                        HidePasswords = false,
+                                                        Manage = true
+                                                    }).Distinct().ToList();
```

Comment:
I'm not sure `Distinct()` is going to have the desired effect here - it'll use the default equality comparer which I assume will be based on reference equality, not value equality.

Can you please check this and also add/update a test to cover the situation where 2 groups provide a manager access to the same collection. (I believe this is the situation it intends to cover?)

Reply:
Distinct() in LINQ compares the property values. I had to include this because we already had one test that failed otherwise `Migrate_CustomUser_WithEditAssignedCollections_WithoutAccessAll_InGroupWithAccessAll_GivesCanManageAccessToAllCollections`.
I have also updated `Migrate_Manager_WithoutAccessAll_GivesCanManageAccess_ToGroupAssignedCollections` to test this.